HIF.md

```diff
@@ -0,0 +1,739 @@
+following variable-length structure:
+
+ - `uint8_t chan_func`  
+    Wi-SUN channel function, supported values are:
+    - `0`: Single fixed channel. (API >= 2.1.1)
+    - `2`: Direct Hash 1 (DH1CF) defined by Wi-SUN.
+
+Only present if `chan_func == 0`:
+
+  - `uint16_t chan_fixed`  
+     Fixed channel number, valid range depends on the selected PHY.
+
+Only present if `chan_func == 2`:
+
+  - `uint8_t chan_mask_len`  
+     Number of bytes in the following channel mask.
+
+  - `uint8_t chan_mask[]`  
+     Bitmask of used channels (1 for used, 0 for excluded). Channels are
+     counted byte by byte, from least to most significant bit (channel `n` maps
+     to `chan_mask[n / 8] & (1 << (n % 8))`). Integrators are responsible for
+     meeting local regulation contraints by excluding disallowed channels from
+     the selected PHY.
+
+[fhss]:     #frequency-hopping-fhss-configuration
+[chan-seq]: #channel-sequence
+[uc]:       #0x30-set_fhss_uc
+[bc]:       #0x31-set_fhss_ffn_bc
+[bc-lfn]:   #0x32-set_fhss_lfn_bc
+[async]:    #0x33-set_fhss_async
+
+### `0x30 SET_FHSS_UC`
+
+Configure unicast schedule for reception.
+
+ - `uint8_t dwell_interval`  
+    Unicast dwell interval in milliseconds (from US-IE).
+
+ - `struct chan_seq`  
+   [Channel sequence][chan-seq] description.
+
+### `0x31 SET_FHSS_FFN_BC`
+
+Configure broadcast schedule for reception and transmission from/to FFN.
+
+ - `uint32_t interval`  
+    Broadcast interval in milliseconds (from BS-IE).
+
+ - `uint16_t bsi`  
+    Broadcast Schedule ID (from BS-IE).
+
+ - `uint16_t dwell_interval`  
+    Broadcast dwell interval in milliseconds (from BS-IE).
+
+ - `struct chan_seq`  
+   [Channel sequence][chan-seq] description.
+
+<!-- TODO: document parent BS-IE following -->
+
+### `0x32 SET_FHSS_LFN_BC`
+
+Configure broadcast schedule for transmission to LFN.
+
+ - `uint16_t interval`  
+    LFN broadcast interval in milliseconds (from LBS-IE).
+
+ - `uint16_t bsi`  
+    LFN broadcast Schedule ID (from LBS-IE).
+
+ - `struct chan_seq`  
+   [Channel sequence][chan-seq] description.
+
+> [!WARNING]
+> The current RCP implementation uses the same set of parameters for FFN and
+> LFN brodcast channel sequences. Thus, [`SET_FHSS_FFN_BC`][bc] and
+> [`SET_FHSS_LFN_BC`][bc-lfn] override each other's channel sequence. The other
+> parameters are distinct.
+
+> [!NOTE]
+> The host is responsible for periodically sending LFN Time Sync frames in
+> order to maintain broadcast timing in LFN children.
+
+### `0x33 SET_FHSS_ASYNC`
+
+Configure asynchronous transmissions for network discovery.
+
+ - `uint32_t tx_duration`  
+    Maximum number of milliseconds the RCP is allowed to stay in continuous TX
+    for an async transmission. If that duration is exceeded, the async
+    transmission is split into chunks of that duration until all channels have
+    been used. By default, the value is set to `0xffffffff`, which disables this
+    feature.
```

Comment:
Should we write about the benefits of doing so?
something along the lines of `allowing RX and TX between chunks`?